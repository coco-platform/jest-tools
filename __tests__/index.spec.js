/**
 * @description - jest-tools suits
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

const path = require('path');

describe('jest-tools suits', () => {
  it('should process static asset', () => {
    // eslint-disable-next-line global-require
    const assetTransformer = require('../lib/asset-transformer');
    const target = path.resolve(__dirname, '__fixture__', 'demo.jpg');
    const result = assetTransformer.process(null, target);

    expect(result).toMatchSnapshot();
  });

  it('should transform status asset within jest', () => {
    // eslint-disable-next-line global-require
    const demo = require('./__fixture__/demo.jpg');

    expect(demo).toMatchSnapshot();
  });

  it('should ignore static css / pcss import', () => {
    // eslint-disable-next-line global-require
    const cssIgnore = require('../lib/css-ignore');

    expect(cssIgnore).toEqual({});
  });

  it('should ignore static css / pcss import within jest', () => {
    // eslint-disable-next-line global-require
    const demo = require('./__fixture__/demo.css');

    expect(demo).toEqual({});
  });

  it('should mapping css modules property', () => {
    // eslint-disable-next-line global-require
    const identityProxy = require('../lib/css-modules');

    expect(identityProxy.hello).toEqual('hello');
  });

  it('should mapping css modules property', () => {
    // eslint-disable-next-line global-require
    const identityProxy = require('../lib/css-modules');

    expect(Reflect.get(identityProxy, '__esModule')).toBeFalsy();
  });

  it('should mapping css modules property within jest', () => {
    // eslint-disable-next-line global-require
    const demo = require('./__fixture__/demo.pcss');

    expect(demo.hello).toEqual('hello');
  });
});
