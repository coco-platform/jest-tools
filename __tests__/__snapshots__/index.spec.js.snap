// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-tools suits should process static asset 1`] = `"module.exports = \\"demo.jpg\\";"`;

exports[`jest-tools suits should transform status asset within jest 1`] = `"demo.jpg"`;
